### PR TITLE
CORE-1667 Clear firmware files on logout

### PR DESCRIPTION
**Pull request by luhmirin-s**

### CODE REVIEW

**Review by alexandr-simprints — APPROVED** [review]

**Review by BurningAXE — APPROVED** [review]

File: fingerprint/infra/scanner/src/main/java/com/simprints/fingerprint/infra/scanner/data/local/FirmwareLocalDataSource.kt [hunk 1]
```diff
@@ -14,34 +17,60 @@ import javax.inject.Inject
...
+
     @Inject
-    constructor(@ApplicationContext context: Context) : this(context, FileUtil)
+    constructor(
+        @ApplicationContext context: Context,
+        @DispatcherIO dispatcher: CoroutineDispatcher,
+    ) : this(context, dispatcher, FileUtil)
 
-    fun getAvailableScannerFirmwareVersions() =
+    fun getAvailableScannerFirmwareVersions() = runBlocking(dispatcher) {
```
[the remark this answers is not in the document]
Comment (luhmirin-s) [reply fingerprint/infra/scanner/src/main/java/com/simprints/fingerprint/infra/scanner/data/local/FirmwareLocalDataSource.kt:30]: To make sure that it runs on IO dispatcher without enforcing `suspend` on the callers for now. I will check how big of a deal that would be, tho.

[the remark this answers is not in the document]
Comment (luhmirin-s) [reply fingerprint/infra/scanner/src/main/java/com/simprints/fingerprint/infra/scanner/data/local/FirmwareLocalDataSource.kt]: It was not a big deal after all 🤷🏻

[the remark this answers is not in the document]
Comment (BurningAXE) [reply fingerprint/infra/scanner/src/main/java/com/simprints/fingerprint/infra/scanner/data/local/FirmwareLocalDataSource.kt]: Wouldn't this allow users to block the main thread - maybe even without realizing it?

[the remark this answers is not in the document]
Comment (luhmirin-s) [reply fingerprint/infra/scanner/src/main/java/com/simprints/fingerprint/infra/scanner/data/local/FirmwareLocalDataSource.kt]: If called on the main thread, probably. Turns out that pretty much all the callers are suspending functions already anyway, so we can just switch dispatchers instead.

**Review by alexandr-simprints — APPROVED** [review]

File: fingerprint/infra/scanner/src/main/java/com/simprints/fingerprint/infra/scanner/data/local/FirmwareLocalDataSource.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (BurningAXE) [reply fingerprint/infra/scanner/src/main/java/com/simprints/fingerprint/infra/scanner/data/local/FirmwareLocalDataSource.kt]: I'd prefer to make it error-proof and not count on the callers knowing how the function should (not) be used. Plus, if they are all suspending anyway, then it should be a problem to make these suspending, too.

[the remark this answers is not in the document]
Comment (luhmirin-s) [reply fingerprint/infra/scanner/src/main/java/com/simprints/fingerprint/infra/scanner/data/local/FirmwareLocalDataSource.kt]: Already did that :D

[the remark this answers is not in the document]
Comment (BurningAXE) [reply fingerprint/infra/scanner/src/main/java/com/simprints/fingerprint/infra/scanner/data/local/FirmwareLocalDataSource.kt]: Ah, sorry :)

**Pull request merged**